opta/commands/apply.py

```diff
@@ -130,6 +130,42 @@ def _apply(
     else:
         raise Exception(f"Cannot handle upload config for cloud {layer.cloud}")
 
+    service_modules = (
+        layer.get_module_by_type("k8s-service")
+        if layer.cloud == "aws"
+        else layer.get_module_by_type("gcp-k8s-service")
+    )
+
+    if len(service_modules) > 0 and (
+        Terraform.downloaded_state.get(layer.name, False)
+        or Terraform.download_state(layer)
+        and get_cluster_name(layer) is not None
```

Comment:
When does this happen? And do we need to handle that?

Reply:
This only happens in the first run with the k8s-service module and given that it can't have preexisting pods (because it is just being deployed now) there's nothing to handle